src/go/rpk/pkg/kafka/utils.go

```diff
@@ -0,0 +1,156 @@
+	"github.com/spf13/afero"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+)
+
+// internal topics
+const CoprocessorTopic = "coprocessor_internal_topic"
+
+func CreateProducer(
+	brokers func() []string, configuration func() (*config.Config, error),
+) func(bool, int32) (sarama.SyncProducer, error) {
+	return func(jvmPartitioner bool, partition int32) (sarama.SyncProducer, error) {
+		conf, err := configuration()
+		if err != nil {
+			return nil, err
+		}
+		cfg, err := LoadConfig(conf)
+		if err != nil {
+			return nil, err
+		}
+		if jvmPartitioner {
+			cfg.Producer.Partitioner = kafkautil.NewJVMCompatiblePartitioner
+		}
+
+		if partition > -1 {
+			cfg.Producer.Partitioner = sarama.NewManualPartitioner
+		}
+
+		return sarama.NewSyncProducer(brokers(), cfg)
+	}
+}
+
+func PublishMessage(
+	producer sarama.SyncProducer,
+	message []byte,
+	key string,
+	topic string,
+	header []sarama.RecordHeader,
+) error {
+
+	k := sarama.StringEncoder(key)
+	ts := time.Now()
+
+	msg := &sarama.ProducerMessage{
+		Topic:		topic,
+		Key:		k,
+		Timestamp:	ts,
+		Value:		sarama.ByteEncoder(message),
+		Headers:	header,
+	}
+
+	retryConf := DefaultConfig().Producer.Retry
+	part, offset, err := RetrySend(
+		producer,
+		msg,
+		uint(retryConf.Max),
+		retryConf.Backoff,
+	)
+	if err != nil {
+		log.Error("Failed to send record")
+		return err
+	}
+
+	log.Infof(
+		"Sent record to partition %d at offset %d with timestamp %v.",
+		part,
+		offset,
+		ts,
+	)
+	return nil
+}
+
+func CreateClient(
+	fs afero.Fs,
+	brokers func() []string,
+	configuration func() (*config.Config, error),
+) func() (sarama.Client, error) {
+	return func() (sarama.Client, error) {
+		conf, err := configuration()
+		if err != nil {
+			return nil, err
+		}
+		bs := brokers()
+		return InitClientWithConf(conf, bs...)
+	}
+}
+
+func CreateAdmin(
+	fs afero.Fs,
+	brokers func() []string,
+	configuration func() (*config.Config, error),
+) func() (sarama.ClusterAdmin, error) {
+	return func() (sarama.ClusterAdmin, error) {
+		conf, err := configuration()
+		if err != nil {
+			return nil, err
+		}
+		cfg, err := LoadConfig(conf)
+		if err != nil {
+			return nil, err
+		}
+		return sarama.NewClusterAdmin(brokers(), cfg)
+	}
+}
+
+func ExistingTopic(admin sarama.ClusterAdmin, topic string) (bool, error) {
+	topics, err := admin.ListTopics()
+	if err != nil {
+		return false, err
+	}
+	if _, val := topics[topic]; val {
+		return true, err
+	} else {
+		return false, err
+	}
+}
+
+/**
+Create coprocessor internal topic with this config:
+	cleanup.policy = compact
+	replicationFactor = if cluster has 2 brokers o more
+						factor is 3, otherwise it should be 1
+	name = coprocessor_internal_topic
+*/
+func CreateCoprocessorTopic(admin sarama.ClusterAdmin) error {
+	brokers, _, err := admin.DescribeCluster()
+	if err != nil {
+		return err
+	}
+	brokersLen := len(brokers)
+	var replicationFactor int16 = 1
+	if brokersLen > 1 {
+		replicationFactor = 3
```

Comment:
I think roko said if we had 7 or more we wanted this to be 5? I forget the details

Reply:
https://vectorizedio.slack.com/archives/C01819S3ARH/p1611077081005200 
the last approach is 1 or 3, I think we need to know how it works on a large cluster and take decition  "how should it grow?"